etrago/tools/io.py

```diff
@@ -788,45 +790,99 @@ def decommissioning(network, session, args, **kwargs):
           
     """  
 
-    if args['gridversion'] == None:   
-        ormclass = getattr(import_module('egoio.db_tables.model_draft'),
-                           'EgoGridPfHvExtensionLine')
-    else:
-        ormclass = getattr(import_module('egoio.db_tables.grid'),
-                           'EgoPfHvExtensionLine')
+    components = ['Line', 'Link', 'Transformer', 'Storage', 'Generator']
+    
+    for comp in components:
 
-    query = session.query(ormclass).filter(
-                        ormclass.scn_name == 'decommissioning_' + 
-                        args['scn_decommissioning'])
+            
+        if version== None:   
+            ormclass = getattr(import_module('egoio.db_tables.model_draft'),
+                           'EgoGridPfHvExtension' + comp)
+        else:
+            ormclass = getattr(import_module('egoio.db_tables.grid'),
+                           'EgoPfHvExtensionLine' + comp)
```

Comment:
in line 800 and 803 it seems wrong that it is '...HvExtension' and '...HvExtensionLine' ?!

Reply:
Yes, that was wrong, I changed it in my latest commit